rest-api/src/main/scala/za/co/absa/enceladus/rest_api/controllers/v3/MappingTableControllerV3.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.rest_api.controllers.v3
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.{HttpStatus, ResponseEntity}
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.security.core.annotation.AuthenticationPrincipal
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.web.bind.annotation._
+import za.co.absa.enceladus.model._
+import za.co.absa.enceladus.rest_api.services.v3.MappingTableServiceV3
+
+import java.util.concurrent.CompletableFuture
+import javax.servlet.http.HttpServletRequest
+
+@RestController
+@RequestMapping(Array("/api-v3/mapping-tables"))
+class MappingTableControllerV3 @Autowired()(mappingTableService: MappingTableServiceV3)
+  extends VersionedModelControllerV3(mappingTableService) {
+
+  import za.co.absa.enceladus.rest_api.utils.implicits._
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  @GetMapping(path = Array("/{name}/{version}/defaults"))
+  @ResponseStatus(HttpStatus.OK)
+  def getDefaults(@PathVariable name: String,
+                  @PathVariable version: String): CompletableFuture[Seq[DefaultValue]] = {
+
+    forVersionExpression(name, version)(mappingTableService.getVersion).map { // "latest" version is accepted
+      case Some(entity) => entity.defaultMappingValue
+      case None => throw notFound()
+    }
+  }
+
+  @PutMapping(path = Array("/{name}/{version}/defaults"))
+  @ResponseStatus(HttpStatus.CREATED)
+  @PreAuthorize("@authConstants.hasAdminRole(authentication)")
+  def updateDefault(@AuthenticationPrincipal user: UserDetails,
```

Comment:
```suggestion
  def updateDefaults(@AuthenticationPrincipal user: UserDetails,
```

As this overwrites the whole map with the map sent in, I think the plural is more understandable.

Reply:
Yes, I agree. Changed as suggested.